lib/elixir/lib/access.ex

```diff
@@ -137,7 +137,8 @@ defmodule Access do
 
   """
 
-  @type t :: list | map | nil | any
+  @type container :: keyword | struct | map | any
+  @type t :: container | nil
```

Comment:
`struct` is to make it more clear that we can have `struct`s in addition to `map`s.
Same as we do with `keyword` and `map` besides having `any`.
But I'm unsure if this is the right approach